opencog/atoms/pattern/PatternLink.cc

```diff
@@ -889,7 +890,9 @@ void PatternLink::make_term_tree_recursive(const Handle& root,
 	_pat.connected_terms_map[{h, root}].emplace_back(ptm);
 
 	// Update the PatternTerm quotation level
-	if (QUOTE_LINK == t)
+	if (not parent->isQuoted() and LOCAL_QUOTE_LINK == t)
+		ptm->setLocalQuote(true);
```

Comment:
On second thought I'm not sure one can do that. What if one uses `UnquoteLink` that is not yet preceded by `QuoteLink`, and in that case we might be unable to use the quotation level to store local quotation. Maybe it's still possible but I'd rather go with the certain route for now, and optimize afterward if possible.